pkg/tsdb/prometheus/buffered/time_series_query.go

```diff
@@ -92,6 +94,17 @@ func New(roundTripper http.RoundTripper, tracer tracing.Tracer, settings backend
 }
 
 func (b *Buffered) ExecuteTimeSeriesQuery(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	// Add headers from the request to context so they are added later on by a context middleware. This is because
+	// prom client does not allow us to do this directly.
+
+	addHeaders := make(map[string]string)
+
+	if req.Headers["FromAlert"] == "true" {
+		addHeaders["FromAlert"] = "true"
```

Comment:
Is this only for the `FromAlert` info? Out of curiosity why is it needed to send to prometheus? Could it be moved to infra code so that it is always sent downstream?

Reply:
Only for the header. Since it was being sent, people have used it to load balance on. So they point their datasource at their LB and use the header to route differently.

Not sure I follow with moving to the infra code? This doesn't seem like the right way to fix it to me, but trying to undo the experienced regression for the user.

Reply:
Yep got it, this works so no issue in merging.

> Not sure I follow with moving to the infra code?

We are already handling headers in grafana backend in general by providing a HTTP client to datasource, so it does not have to be handled custom, which is why this code was removed. We may somehow move this to the HTTP client that is provided so that `FromAlert` header would be passed upstream for all datasources automatically if it is valuable in general.